src/aihawk_easy_applier.py

```diff
@@ -839,24 +833,34 @@ def _save_questions_to_json(self, question_data: dict) -> None:
         output_file = 'answers.json'
         question_data['question'] = self._sanitize_text(question_data['question'])
 
-        logger.debug(f"Saving question data to JSON: {question_data}")
+        logger.debug(f"Checking if question data already exists: {question_data}")
         try:
-            try:
-                with open(output_file, 'r') as f:
-                    try:
-                        data = json.load(f)
-                        if not isinstance(data, list):
-                            raise ValueError("JSON file format is incorrect. Expected a list of questions.")
-                    except json.JSONDecodeError:
-                        logger.error("JSON decoding failed")
-                        data = []
-            except FileNotFoundError:
-                logger.warning("JSON file not found, creating new file")
-                data = []
-            data.append(question_data)
+            with open(output_file, 'r+') as f:
+                try:
+                    data = json.load(f)
+                    if not isinstance(data, list):
+                        raise ValueError("JSON file format is incorrect. Expected a list of questions.")
+                except json.JSONDecodeError:
+                    logger.error("JSON decoding failed")
+                    data = []
+
+                # Check if the question already exists
+                question_exists = any(item['question'] == question_data['question'] for item in data)
```

Comment:
this can be moved to a function.
any(item['question'] == question_data['question'] for item in data)

Reply:
done